### PR TITLE
new(modern_probe): Add mmap, munmap

### PR DESCRIPTION
**Pull request by dwindsor**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

/area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

#723 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/modern_bpf/test_suites/syscall_enter_suite/mmap_e.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: addr (type: PT_UINT64) */
+	evt_test->assert_only_param_len(1, sizeof(uint64_t));
```
Comment (dwindsor) [inline test/modern_bpf/test_suites/syscall_enter_suite/mmap_e.cpp:42]: Impossible to know the value of this parameter deterministically (it is an address).

File: test/modern_bpf/test_suites/syscall_exit_suite/mmap_x.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO)*/
+	evt_test->assert_numeric_param(1, (int64_t)errno_value);
+
+	/* Parameter 2: vm_size (type: PT_UINT32) */
+	evt_test->assert_only_param_len(2, sizeof(uint32_t));
```
Comment (dwindsor) [inline test/modern_bpf/test_suites/syscall_exit_suite/mmap_x.cpp:46]: Impossible to know these values.

File: test/modern_bpf/test_suites/syscall_enter_suite/mmap_e.cpp (hunk 1 above)
Comment (Andreagit97) [reply test/modern_bpf/test_suites/syscall_enter_suite/mmap_e.cpp]: here we can check against the address we used in the syscall

**Comment by Andreagit97** [comment]
This PR LGTM! Amazing work as always! :partying_face: I've just applied some suggestions feel free to revert them :)

**Comment by Andreagit97** [comment]
Ei @dwindsor any news on this PR and https://github.com/falcosecurity/libs/pull/806? They were almost complete if you want I can take care of rebasing and merging them :)

**Comment by dwindsor** [comment]
> Ei @dwindsor any news on this PR and #806? They were almost complete if you want I can take care of rebasing and merging them :)

So sorry I had to take a break from this! I just rebased, is there anything else left to do? =)

**Comment by Andreagit97** [comment]
> So sorry I had to take a break from this! I just rebased, is there anything else left to do? =)

Hey @dwindsor don't worry at all! The issue here is that since your last changes, things are changed a little bit, now driver tests cover all the 3 drivers not only the modern probe. In this case, the modern probe implementation is correct but the old bpf probe `mmap` test is failing, and we cannot merge the PR until all tests are green :( Same for the `read` PR. If you want I can fix them, let me know :)

**Comment by dwindsor** [comment]
> Hey @dwindsor don't worry at all! The issue here is that since your last changes, things are changed a little bit, now driver tests cover all the 3 drivers not only the modern probe. In this case, the modern probe implementation is correct but the old bpf probe mmap test is failing, and we cannot merge the PR until all tests are green :( Same for the read PR. If you want I can fix them, let me know :)

Yes, please! That would be very helpful - that way I can see the new way it's done and maybe add some more later =)

**Comment by Andreagit97** [comment]
Rebased and fixed old drivers' test failures :) Now I will go with #806 and #900

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by dwindsor** [comment]
> Rebased and fixed old drivers' test failures :) Now I will go with #806 and #900

Thanks for taking care of this!

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**